scipy/special/tests/test_trig.py

```diff
@@ -57,8 +56,123 @@ def test_intermediate_overlow():
         assert_allclose(res.real, std.real)
         assert_allclose(res.imag, std.imag)
 
+def test_sinpi():
+    x = np.arange(-16, 16, 1 / 16)
 
-def test_zero_sign():
+    expected = np.sin(np.pi * x)
+    actual = sinpi(x)
+
+    assert_allclose(expected, actual, atol=1e-10)
```

Comment:
The function `assert_allclose` takes `actual` first and `expected` second. So it would be better to swap your arguments above so that the relative error is computed correctly.

Separately, the default relative error tolerance of `assert_allclose` is rather high (`1e-7`). For these functions I'd suggest to set `rtol` to no larger than `1e-15`; the actual relative error is likely below `3e-16`.